test/unit_test.c

```diff
@@ -2184,16 +2185,23 @@ static void test_util(void) {
 
   {
     extern bool mg_to_size_t(struct mg_str, size_t *);
-    size_t val = 1;
+    size_t val;
     ASSERT(mg_to_size_t(mg_str("0"), &val) && val == 0);
     ASSERT(mg_to_size_t(mg_str("123"), &val) && val == 123);
-    ASSERT(mg_to_size_t(mg_str(""), &val) && val == 0);
+    ASSERT(mg_to_size_t(mg_str(" 123 \t"), &val) && val == 123);
+    ASSERT(mg_to_size_t(mg_str(""), &val) == false);
+    ASSERT(mg_to_size_t(mg_str(" 123x"), &val) == false);
     ASSERT(mg_to_size_t(mg_str("-"), &val) == false);
+    mg_snprintf(buf, sizeof(buf), "%lu", SIZE_MAX);
+    MG_INFO(("--> %s", buf));
+    ASSERT(mg_to_size_t(mg_str(buf), &val) && val == SIZE_MAX);
+#if 0
     ASSERT(mg_to_size_t(mg_str("18446744073709551616"), &val) ==
            false);  // range +1
     ASSERT(mg_to_size_t(mg_str("18446744073709551610"), &val) == false);
     // TODO(): ASSERT(mg_to_size_t(mg_str("18446744073709551609"), &val) &&
     //         val == 18446744073709551609U);  // our max or SIZE_MAX
+#endif
```

Comment:
These can go now
Yeah, SIZE_MAX and company cause headaches in some platforms...